llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -4976,11 +4973,7 @@ LoopVectorizationCostModel::selectInterleaveCount(ElementCount VF,
       MaxInterleaveCount = ForceTargetMaxVectorInterleaveFactor;
   }
 
-  unsigned EstimatedVF = VF.getKnownMinValue();
-  if (VF.isScalable()) {
-    if (std::optional<unsigned> VScale = getVScaleForTuning(TheLoop, TTI))
-      EstimatedVF *= *VScale;
-  }
+  unsigned EstimatedVF = getEstimatedRuntimeVF(TheLoop, TTI, VF);
   assert(EstimatedVF >= 1 && "Estimated VF shouldn't be less than 1");
```

Comment:
Does it make any sense to move this assert into `getEstimatedRuntimeVF` ?

Reply:
It's a good question. It looks like it was really just asserting that `getVScaleForTuning` returned a non-zero value. Maybe it is worth moving into `getEstimatedRuntimeVF`.